### PR TITLE
Remove panics, add errors

### PR DESCRIPTION
**Pull request by asalzmann**
**Summary**
Per this thread: https://github.com/cosmos/cosmos-sdk/issues/11600 panics should only be used when a golang error (program halting) is about to occur, like a divide by zero or OOB error.

This PR removes all panics given we're misusing them (and a panic in an BeginBlocker / epoch will crash the chain - some of these panics were called by functions in epochs!)

**Test plan**
Integration tests

### CODE REVIEW

**Review by sampocs — COMMENTED** [review]
Looks good! 

One thing I was discussing with Riley yesterday that I think is worth being brought up again - in all these cases we generally override each error and pass a new one. 

Ex: 
```
err := someFunctionCall() // some more specific error about what actually went wrong
if err != nil {
   return sdkerrors...  // some new, but less specific error that we actually return (hiding the one above)
}
```
Should we be actually wrapping the original error instead? 
```
err := someFunctionCall() // some more specific error about what actually went wrong
if err != nil {
   return sdkerrors.SomeNewWrapperError(err) // takes original error as param
}
```

Sounds like this is common practice in cosmos though, but I just want to confirm

File: x/stakeibc/keeper/msg_server_liquid_stake.go [hunk 1]
```diff
@@ -106,22 +106,23 @@ func (k msgServer) MintStAsset(ctx sdk.Context, sender sdk.AccAddress, amount in
...
-	// It will panic if the module account does not exist or is unauthorized.
+	// It will error if the module account does not exist or is unauthorized.
 	sdkerror := k.bankKeeper.MintCoins(ctx, types.ModuleName, stCoins)
 	if sdkerror != nil {
 		k.Logger(ctx).Info("Failed to mint coins")
-		panic(sdkerror)
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to mint coins")
 	}
 	// transfer those coins to the user
 	sdkerror = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, sender, stCoins)
```
Comment (sampocs) [inline x/stakeibc/keeper/msg_server_liquid_stake.go:121]: nit: probably want to assign this to `err` instead of `sdkerror`

**Comment by riley-stride** [comment]
> Looks good!
> 
> One thing I was discussing with Riley yesterday that I think is worth being brought up again - in all these cases we generally override each error and pass a new one.
> 
> Ex:
> 
> ```
> err := someFunctionCall() // some more specific error about what actually went wrong
> if err != nil {
>    return sdkerrors...  // some new, but less specific error that we actually return (hiding the one above)
> }
> ```
> 
> Should we be actually wrapping the original error instead?
> 
> ```
> err := someFunctionCall() // some more specific error about what actually went wrong
> if err != nil {
>    return sdkerrors.SomeNewWrapperError(err) // takes original error as param
> }
> ```
> 
> Not sure if there's a way to wrap the error like that (or maybe I'm misunderstanding and this wrapping happens automatically?). Either way - sounds like this is common practice in cosmos, but just want to confirm

Well phrased. I'd vote we wither return `err` or something like the following, where we have an error type, a descriptive string, as well as `err`: 
```
return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to SubmitTxs for %s, %s, %s | err=%v", zone.ConnectionId, zone.ChainId, msgs, err)
```

**Comment by riley-stride** [comment]
Looks good to me, thanks for actioning this!

**Review by riley-stride — APPROVED** [review]
Looks good to me, thanks for actioning this! Only thing worth looking at before merge IMO is Sam's comment re: error conventions and my reply. Might be quick to reformat all our errors to that convention, but I'm not sure it's the best use of time right now, nor that we want to run with that convention. Wdyt?

**Comment by sampocs** [comment]
> Looks good to me, thanks for actioning this! Only thing worth looking at before merge IMO is Sam's comment re: error conventions and my reply. Might be quick to reformat all our errors to that convention, but I'm not sure it's the best use of time right now, nor that we want to run with that convention. Wdyt?

File: x/stakeibc/keeper/msg_server_liquid_stake.go [hunk 2]
```diff
@@ -106,22 +106,23 @@ func (k msgServer) MintStAsset(ctx sdk.Context, sender sdk.AccAddress, amount in
...
 
 	// mint new coins of the asset type
 	// MintCoins creates new coins from thin air and adds it to the module account.
-	// It will panic if the module account does not exist or is unauthorized.
+	// It will error if the module account does not exist or is unauthorized.
 	sdkerror := k.bankKeeper.MintCoins(ctx, types.ModuleName, stCoins)
 	if sdkerror != nil {
 		k.Logger(ctx).Info("Failed to mint coins")
-		panic(sdkerror)
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to mint coins")
```
[the remark this answers is not in the document]
Comment (asalzmann) [reply x/stakeibc/keeper/msg_server_liquid_stake.go:118]: Agreed, I haven't been diligent about adding new error codes - if we go the route of surfacing specific errors for each function, we should make this meaningful!

**Comment by asalzmann** [comment]
Let's fix these progressively! Does the following achieve both the goal of passing the low-level error up the stack, as well as surfacing something more meaningful in relation to how the enclosing function is erroring?
```
func (k Keeper) DoSomethingSpecific {
    err = k.someKeeperFunction(ctx, ...)
    if err != nil {
	    k.Logger(ctx).Info("Failed to DoSomethingSpecific")
	    return sdkerrors.Wrapf(err, "Failed to DoSomethingSpecific")
    }
}
```

The only advantage of surfacing a custom error is that it has an attached error code, which might be easier for external services to parse than an arbitrary string

Wdyt?

**Comment by sampocs** [comment]
Yup, I think that's perfect! 

The way I've been thinking about it, is there are some cases where we identify the error ourselves, and there are cases where we receive an error from a function we called. I think custom errors make sense in the first case, but less so in the second. 

Ex: 
```
someSpecificConditionWeCheck := ...
if !someSpecificConditionWeCheck {
    return CustomError
}
```
vs 
```
err := someFunctionCall()
if err := nil {
   return WrappedError(err)
}
```

**Comment by sampocs** [comment]
That said, I think most of the cases I've seen are covered with some of the provided sdkerrors out of the box. One I noticed this morning was:
```
strideEpochTracker, found := k.GetEpochTracker(ctx, epochtypes.STRIDE_EPOCH)
if !found {
	k.Logger(ctx).Info("failed to find epoch")
	return nil, sdkerrors.Wrapf(types.ErrInvalidLengthEpochTracker, "no number for epoch (%s)", epochtypes.STRIDE_EPOCH)
}
```
Where we could instead do:
```
strideEpochTracker, found := k.GetEpochTracker(ctx, epochtypes.STRIDE_EPOCH)
if !found {
	k.Logger(ctx).Info("failed to find epoch")
	return nil, sdkerrors.Wrapf(sdkerrors.ErrNotFound, "no number for epoch (%s)", epochtypes.STRIDE_EPOCH)
}
```

Although I hear ya on the event code. I think if we make the wrapped message specific enough though, it should be easy enough to find?

**Comment by sampocs** [comment]
Also (side note) we should make sure we're logging these with an Error status instead of Info

**Comment by asalzmann** [comment]
Good call, updated Infos to Errors where appropriate

**Pull request merged**